packages/samples/headless-react/src/components/facet/facet.class.tsx

```diff
@@ -0,0 +1,68 @@
+import {Component} from 'react';
+import {
+  buildFacet,
+  Facet as HeadlessFacet,
+  FacetActions,
+  FacetState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface FacetProps {
+  field: string;
+  facetId: string;
+}
+
+export class Facet extends Component<FacetProps> {
+  private controller: HeadlessFacet;
+  public state: FacetState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: FacetProps) {
+    super(props);
+    if (process.env.NODE_ENV === 'development') {
+      // When `React.StrictMode` is active in development, the constructor is executed twice.
+      // This ensures the facet isn't already defined from a previous execution.
+      engine.dispatch(FacetActions.removeFacet(this.props.facetId));
+    }
```

Comment:
Unfortunately, it looks like after the first execution of the constructor, every private attribute is set to `undefined`.